File: frontend/src/api/contract.ts
import axiosInstance from "../lib/axiosInstance";
import { QueryRequest, QueryResponse } from "../types/contract";

const BASE_URL = "/api/v1/contract/analyse";

export const analyzeContract = async (
  payload: QueryRequest,
): Promise<QueryResponse> => {
  const { data } = await axiosInstance.post(BASE_URL, payload);
  return data;
};

File: backend/app/api/v1/contract.py
from fastapi import APIRouter, HTTPException
from app.schemas.contract import QueryRequest, QueryResponse
from app.rag.retriever import retrieve
from app.llm.contract_call import generate_response

router = APIRouter()

retriever = retrieve(namespace= "contract")

@router.post('/analyse', response_model=QueryResponse)
async def generate_answer(payload: QueryRequest):
    try:
        
        message = generate_response(payload.query, retriever)
        
        return QueryResponse(answer=message)
    except Exception as e:
        print(e)
        raise HTTPException(status_code=500, detail=str(e))
        



